### PR TITLE
[Clubhouse] Add extractor

### PR DESCRIPTION
**Pull request by Ashish0804**
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

DESCRIPTION

Closes https://github.com/yt-dlp/yt-dlp/issues/4172

@codenamedpkt if u want more fields then provide an english url, otherwise this extractor is good to merge as it is.

### CODE REVIEW

File: yt_dlp/extractor/clubhouse.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        'url': 'https://www.clubhouse.com/room/PN50O4GG',
+        'info_dict': {
+            'id': 'PN50O4GG',
+            'ext': 'ts',
+            'title': '香港消費券轉會！八達通、PayMe、BoC Pay、Tap＆Go、AlipayHK、WeChat Pay 邊個最有伏'
+        }
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
```
Comment (Lesmiscore) [inline yt_dlp/extractor/clubhouse.py:17]: I think this field should be renamed to something else since there's `id` in builtins :sweat_smile:

Comment (Ashish0804) [reply yt_dlp/extractor/clubhouse.py:17]: I haven't been following core changes for past 6+ months...are there any major changes?
Bcoz i've always used `id` in all my extractors 🤔

Comment (Lesmiscore) [reply yt_dlp/extractor/clubhouse.py:17]: nothing stops from using `id` variable at least

Comment (pukkandan) [reply yt_dlp/extractor/clubhouse.py:17]: Generally it's bad to shadow builtins. However the `id` builtin is rarely used. So I generally don't bother correcting this.

(I personally prefer to use `item_id`/`video_id`/`display_id` though)

Comment (Ashish0804) [reply yt_dlp/extractor/clubhouse.py:17]: if it is not a breaking change then I'd rather keep it id only. The key in return dict is `id`, so it makes way more sense to call the var `id` from my perspective. 

I use item_id or video_id in playlist extractors and display_id is a different key all together so it doesn't make sene to me to use those.

Marking this thread resolved.

**Comment by codenamedpkt** [comment]
@Ashish0804 ,
As you sorry to closed PR can you continue with your Clubhouse extractor development so not me only can download the Clubhouse content.

Thanks!

Alice,
Mr. Poon's Studio

ps.: As I found I can run a Windows Server, aka Microsoft Server, with GitHub Actions, I may busy and cannot reply ASAP. But if I free, I will reply all at once.

**Comment by Lesmiscore** [comment]
@codenamedpkt 
You should. He resigned from yt-dlp development. We won't open this PR, anymore.

**Comment by codenamedpkt** [comment]
@Lesmiscore Oh what the guy he resigned! Can you and developers post the request to who willing finish the Clubhouse extractor as I post the new verbose output today at #4172.

**Comment by pukkandan** [comment]
> Can you and developers post the request

You realize we are all volunteers, right? Anyone interested is free to make a PR

**Pull request closed without merging**